Sources/StreamVideoSwiftUI/Utils/ReusePool.swift

```diff
@@ -30,10 +30,12 @@ final class ReusePool<Element: AnyObject & Hashable> {
         self.initialCapacity = initialCapacity
         self.factory = factory
 
-        // Initialize the pool with a set number of elements
-        for _ in 0..<initialCapacity {
-            let element = factory()
-            available.append(element)
+        // Initialize the pool with a set number of elements (thread-safe)
+        queue.sync {
```

Comment:
Hmm i'm not sure about this one. Even without the sync, there aren't multiple threads accessing the initialiser at the same (if they were they would be creating different instances).